source/transfer_list.rst

```diff
@@ -657,5 +659,121 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+.. _spmc_manifest_entry:
+
+SPMC manifest entry layout (XFERLIST_SPMC_MANIFEST)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the SPMC (Secure Partition Manager Core) manifest image.
+This manifest contains the SPMC attribute node consumed by the SPMD
+(Secure Partition Manager Dispatcher) at boot time.
+
+.. _tab_spmc_manifest:
+.. list-table:: SPMC manifest type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **5**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of SPMC manifest in bytes.
+
+   * - spmc_manifest
+     - data_size
+     - hdr_size
+     - Holds a SPMC manifest image.
+
+.. _optee_entries:
+
+Entries defined for OP-TEE
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following entry types are defined for OP-TEE:
+
+**OP-TEE pagable part entry layout (XFERLIST_OPTEE_PAGABLE_PART)**
+
+This entry type holds the address of OP-TEE pagable part.
+This address (of type 'uint64_t') is used when OPTEED (OP-TEE Dispatcher)
+is the Secure Payload Dispatcher, indicating where to load the pagable image of
+OP-TEE OS.
+
+.. _tab_optee_pagable_part:
+.. list-table:: OP-TEE pagable part type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x20**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of OP-TEE pagable part in bytes must be set to **8**.
+
+   * - pagable_part
+     - data_size
+     - hdr_size
+     - Pagable part address for OPTEED
+
+**OPTEED AARCH entry layout (XFERLIST_OPTEED_AARCH)**
+
+This entry type holds the AARCH mode for OPTEED (OP-TEE Dispatcher).
+This AARCH mode (of type 'uint32_t', **0** for AARCH64 and **1** for AARCH32)
```

Comment:
Is this different from the current CPU mode? I would have thought that you could just check the CPU flags, or even hard-code it since the build system knows the code it is building for.

So please could you add in an explanation for why this is needed?

Reply:
The reason I am doing in this way is because the current "legacy" handoff from BL2 and opteed is passing the AARCH mode via X0/R0, but not directly reading from SPSR.
If you agree, I can remove this TE from the spec, and then modify the code to allow opteed to get the AARCH mode from SPSR instead of X0/R0 from BL2.

Reply:
> The reason I am doing in this way is because the current "legacy" handoff from BL2 and opteed is passing the AARCH mode via X0/R0, but not directly reading from SPSR. If you agree, I can remove this TE from the spec, and then modify the code to allow opteed to get the AARCH mode from SPSR instead of X0/R0 from BL2.

I see. Adding this explanation would help, i.e. the first sentence you wrote

Reply:
> If you agree, I can remove this TE from the spec, and then modify the code to allow opteed to get the AARCH mode from SPSR instead of X0/R0 from BL2.

I think that's for the OP-TEE maintainers to agree, not the spec maintainers. Although it sounds tempting to read SPSR, I think AArch64 OPTEED may need to support launching AArch32 OP-TEE. Having said that, recent Arm CPUs don't support AArch32 at ELs > 0.

Reply:
Yes indeed, so just add the explanation and call it done?

Reply:
> > If you agree, I can remove this TE from the spec, and then modify the code to allow opteed to get the AARCH mode from SPSR instead of X0/R0 from BL2.
> 
> I think that's for the OP-TEE maintainers to agree, not the spec maintainers. Although it sounds tempting to read SPSR, I think AArch64 OPTEED may need to support launching AArch32 OP-TEE. Having said that, recent Arm CPUs don't support AArch32 at ELs > 0.

Currently the opteed already holds the value of SPSR within the entry point info structure. So basically we can deprecate passing from the BL2 arguments. I already commit my changes for opteed and OP-TEE maintainers will review it.
For the spec, I think I can remove this TE from now.